### ISSUE TITLE
Getting an error with Angular5 

### ISSUE THREAD

**Issue by xyrintech** [optimistex/ng2-select-ex#4]
StaticInjectorError[DomSanitizer]: 
  StaticInjectorError[DomSanitizer]: 
    NullInjectorError: No provider for DomSanitizer!
    at _NullInjector.get (core.js:923)
    at resolveToken (core.js:1211)
    at tryResolveToken (core.js:1153)
    at StaticInjector.get (core.js:1024)
    at resolveToken (core.js:1211)
    at tryResolveToken (core.js:1153)
    at StaticInjector.get (core.js:1024)
    at resolveNgModuleDep (core.js:10585)
    at NgModuleRef_.get (core.js:11806)
    at resolveNgModuleDep (core.js:10585)

**Comment by optimistex** [comment]
Give me more details to reproduce the bug.

**Comment by optimistex** [comment]
I checked it on an empty project with Angular5 and don't found the bug.

```console
Angular CLI: 1.6.6
Node: 9.4.0
OS: linux x64
Angular: 5.2.3
```
______
src/app/app.module.ts
```typescript
import {BrowserModule} from '@angular/platform-browser';
import {NgModule} from '@angular/core';
import {AppComponent} from './app.component';
import {NgxSelectModule} from 'ng2-select-ex';
@NgModule({
    declarations: [
        AppComponent
    ],
    imports: [
        BrowserModule,
        NgxSelectModule
    ],
    providers: [],
    bootstrap: [AppComponent]
})
export class AppModule {
}
```
______________
src/app/app.component.ts
```typescript
import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  title = 'app';
}
```
______________
src/app/app.component.html
```html
<ngx-select [items]="['sdfsd','dsgdg']"></ngx-select>
```
______________
src/index.html
```html
<!doctype html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>TestProj</title>
  <base href="/">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="icon" type="image/x-icon" href="favicon.ico">
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
</head>
<body>
  <app-root></app-root>
</body>
</html>
```
______________
Result:
![image](https://user-images.githubusercontent.com/11228640/35737342-3684ca06-083c-11e8-9f2d-5e6ecd0db4a5.png)

**Comment by xyrintech** [comment]
I am still getting this error, please check this out: https://drive.google.com/file/d/1Wy2-xXuN3jkx2GIe-tsPQ_VaM6z4hwsh/view

**Comment by xyrintech** [comment]
I have imported the NgxSelectModule in a shared module.
Sharing the code wont be possible as this is a huge project and replicating it on a core project wont be possible. Hope you understand

**Comment by optimistex** [comment]
I have managed to get the error:
```console
Error: StaticInjectorError(AppModule)[ApplicationRef -> NgZone]: 
  StaticInjectorError(Platform: core)[ApplicationRef -> NgZone]: 
    NullInjectorError: No provider for NgZone!
    at _NullInjector.get (core.js:994)
    at resolveToken (core.js:1292)
    at tryResolveToken (core.js:1234)
    at StaticInjector.get (core.js:1102)
    at resolveToken (core.js:1292)
    at tryResolveToken (core.js:1234)
    at StaticInjector.get (core.js:1102)
    at resolveNgModuleDep (core.js:10837)
    at _createClass (core.js:10882)
    at _createProviderInstance$1 (core.js:10848)
```

Currently, I am working on it...

**Comment by xyrintech** [comment]
Sounds good. Thanks.

**Comment by optimistex** [comment]
I did figure out a solution.
The `StaticInjectorError` seems to the trouble of the Angular 5.

The issue appearing when a package installed by `npm link ...`, by way for my case.
For solving you have to use the "preserve-symlinks". Exists two ways for do it:
1. run with flag `--preserve-symlinks` for instance `ng serve --preserve-symlinks`: 
2. enlarge your `.angular-cli.json`: https://github.com/angular/angular/issues/20598#issuecomment-347629172
```json
"defaults": {
    "styleExt": "css",
    "component": {},
    "build": {
        "preserveSymlinks": true
    }
}
```

**Comment by xyrintech** [comment]
Issue still exist :)

**Comment by xyrintech** [comment]
I have used ng serve --preserve-symlinks

**Comment by optimistex** [comment]
Can you make some project for example with the bug? Maybe a little example on the planker!?
I have no more ideas what are you doing for catching that :(

**Comment by KevinJTaylor** [comment]
I'm seeing the same issue, but I'm not using ng serve. I do an ng build and then serve my app via an Express server. The app itself is too big to even think about posting on here, but I have chanced upon something which I just don't understand. A colleague pulled the same codebase out of GIT and did a clean/build on everything (we're using yarn/lerna), and she does not see the same issue. 

I guess there is a chance that she is running slightly different versions of something dependency, but I do try to use absolute versions in all package.json files, so it is unlikely. I've also tried running in a different browser, wondering whether something was cached, but no luck. I'm starting to think it is a timing/synchronisation thing although that is a guess based on very little evidence.
